src/persist-client/src/internal/trace.rs

```diff
@@ -369,22 +369,19 @@ impl<T> Trace<T> {
     }
 
     pub fn map_batches<'a, F: FnMut(&'a HollowBatch<T>)>(&'a self, mut f: F) {
-        self.spine.map_batches(move |b| match b {
-            SpineBatch::Merged(b) => f(&b.batch),
-            SpineBatch::Fueled { parts, .. } => {
-                for b in parts.iter() {
-                    f(&b.batch);
-                }
-            }
-        })
+        for batch in self.batches() {
+            f(batch);
+        }
     }
 
-    #[must_use]
-    pub fn batches(&self) -> impl IntoIterator<Item = &HollowBatch<T>> {
-        // It should be possible to do this without the Vec.
-        let mut batches = Vec::new();
-        self.map_batches(|b| batches.push(b));
-        batches
+    pub fn batches(&self) -> impl Iterator<Item = &HollowBatch<T>> {
+        self.spine
+            .spine_batches()
+            .flat_map(|b| match b {
+                SpineBatch::Merged(b) => std::slice::from_ref(b),
```

Comment:
TIL about `std::slice::from_ref`

Reply:
Unironically one of my favourite rust functions. 😬 Comes up surprisingly often!